### PR TITLE
Replace Opentracing with OTel

### PR DESCRIPTION
**Pull request by SylvainJuge**
- migration OpenT{racing,elemetry}
- remove useless jar packaging (default)

### CODE REVIEW

File: opbeans/pom.xml [hunk 1]
```diff
@@ -39,9 +38,9 @@
 			<version>1.28.4</version>
 		</dependency>
 		<dependency>
-			<groupId>co.elastic.apm</groupId>
-			<artifactId>apm-opentracing</artifactId>
-			<version>1.3.0</version>
+			<groupId>io.opentelemetry</groupId>
+			<artifactId>opentelemetry-api</artifactId>
+			<version>1.9.1</version>
```
Comment (jlvoiseux) [inline opbeans/pom.xml:43]: Seems like the [latest](https://github.com/open-telemetry/opentelemetry-java) available version is v1.11.0. Should we take this into account ?

**Review by eyalkoren — APPROVED** [review]

**Review by jlvoiseux — APPROVED** [review]

File: opbeans/src/main/java/co/elastic/apm/opbeans/controllers/APIRestController.java [hunk 2]
```diff
@@ -59,44 +59,46 @@
...
         this.customerRepository = customerRepository;
         this.orderRepository = orderRepository;
-        this.tracer = new ElasticApmTracer();
     }
 
     @GetMapping(value = "/products")
     @CaptureSpan("Annotation products span")
     Collection<ProductList> products() {
-        ElasticApm.currentSpan().addTag("foo", "bar");
+        ElasticApm.currentSpan().setLabel("foo", "bar");
```
[the remark this answers is not in the document]
Comment (SylvainJuge) [reply opbeans/src/main/java/co/elastic/apm/opbeans/controllers/APIRestController.java:83]: The goal here was not to remove entirely the Elastic API usage by OTel one, but to keep entry points to test both.

File: opbeans/pom.xml [hunk 3]
```diff
@@ -39,9 +38,9 @@
 			<version>1.28.4</version>
 		</dependency>
 		<dependency>
-			<groupId>co.elastic.apm</groupId>
-			<artifactId>apm-opentracing</artifactId>
-			<version>1.3.0</version>
+			<groupId>io.opentelemetry</groupId>
```
[the remark this answers is not in the document]
Comment (SylvainJuge) [reply opbeans/pom.xml:41]: Good point for the BOM, I will have a look at it in a follow-up PR. Here the goal was just to import the API to keep it a minimal change.

Maybe using a maven profile to optionally include the SDK would also be relevant here, so combined with work by @jlvoiseux  on the matter we could easily test the following deployment options:
- OTel API and Elastic agent API are included in app, and default to no-op unless OTel SDK, OTel agent or Elastic agent is added
- App with OTel SDK : would require to rebuild to include optional dependencies, unless there i a way to include the SDK dependencies but disable it.
- ability to test all of this with and without OTel and Elastic agents

File: opbeans/src/main/java/co/elastic/apm/opbeans/controllers/APIRestController.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (cyrille-leclerc) [reply opbeans/src/main/java/co/elastic/apm/opbeans/controllers/APIRestController.java:83]: ack, would a comment in the code help people understand the intent here?
We don't want people who want to choose a plain Otel API instrumentation to misunderstand and have the feeling that Elastic Otel API support requires some proprietary code as well.

File: opbeans/pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (cyrille-leclerc) [reply opbeans/pom.xml:41]: Do we want:
* Our Elastic annotations to be recognized by Otel agent? See https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/v1.10.1/instrumentation/external-annotations/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/extannotations/ExternalAnnotationInstrumentation.java
* Our Elastic API calls to be recognized by the OTel agent, similar to the [OpenTelemetry Jaeger Shim](https://github.com/open-telemetry/opentelemetry-java/tree/main/opentracing-shim)?

It's an interesting question, I'm not sure of the answer in terms of prioritization.

[the remark this answers is not in the document]
Comment (felixbarny) [reply opbeans/pom.xml:41]: We could also add support for [`@WithSpan`](https://github.com/open-telemetry/opentelemetry-java/blob/main/extensions/annotations/src/main/java/io/opentelemetry/extension/annotations/WithSpan.java) in the Elastic Java agent.

**Comment by SylvainJuge** [comment]
@cyrille-leclerc FYI I've started adding SDK + auto-configure in a draft PR : https://github.com/elastic/opbeans-java/pull/110

**Pull request merged**